google/cloud/forseti/services/inventory/base/gcp.py

```diff
@@ -33,82 +34,434 @@
+        """
+
+    @abc.abstractmethod
+    def iter_gae_versions(self, projectid, serviceid):
+        """Iterate gae versions from GCP API.
+
+        Args:
+            projectid (str): id of the project to query
+            serviceid (str): id of the appengine service
+        """
+
+    @abc.abstractmethod
+    def iter_gae_instances(self, projectid, serviceid, versionid):
+        """Iterate gae instances from GCP API.
+
+        Args:
+            projectid (str): id of the project to query
+            serviceid (str): id of the appengine service
+            versionid (str): id of the appengine version
+        """
+
+    @abc.abstractmethod
+    def iter_container_clusters(self, projectid):
+        """Iterate Kubernetes Engine Cluster from GCP API.
 
         Args:
             projectid (str): id of the project to query
+        """
 
-        Raises:
-            NotImplementedError: Because not implemented.
+    @abc.abstractmethod
+    def fetch_container_serviceconfig(self, projectid, zone=None,
+                                      location=None):
+        """Fetch Kubernetes Engine per zone service config from GCP API.
+
+        Args:
+            projectid (str): id of the project to query
+            zone (str): zone of the Kubernetes Engine
+            location (str): location of the Kubernetes Engine
         """
-        raise NotImplementedError()
 
+    @abc.abstractmethod
+    def iter_buckets(self, projectid):
+        """Iterate Buckets from GCP API.
+
+        Args:
+            projectid (str): id of the project to query
+        """
+
+    @abc.abstractmethod
     def iter_objects(self, bucket_id):
-        """Not Implemented.
+        """Iterate Objects from GCP API.
 
         Args:
             bucket_id (str): id of the bucket to get
+        """
+
+    @abc.abstractmethod
+    def iter_datasets(self, projectid):
+        """Iterate Datasets from GCP API.
 
-        Raises:
-            NotImplementedError: Because not implemented.
+        Args:
+            projectid (str): id of the project to query
         """
-        raise NotImplementedError()
 
-    def get_organization_iam_policy(self, orgid):
-        """Not Implemented.
+    @abc.abstractmethod
+    def iter_cloudsqlinstances(self, projectid):
+        """Iterate Cloud sql instances from GCP API.
+
+        Args:
+            projectid (str): id of the project to query
+        """
+
+    @abc.abstractmethod
+    def is_compute_api_enabled(self, projectid):
+        """Verifies the Compute API is enabled on a project.
+
+        Args:
+            projectid (str): id of the project to query
+        """
+
+    @abc.abstractmethod
+    def fetch_compute_project(self, projectid):
+        """Fetch compute project data from GCP API.
+
+        Args:
+            projectid (str): id of the project to query
+        """
+
+    @abc.abstractmethod
+    def iter_computeinstances(self, projectid):
```

Comment:
nit: There is a bit of inconsistency of how the words are spaced here.  `fetch_compute_project` and `iter_computeinstance`.  Would prefer spacing every word individually: `iter_compute_instance`